jax/lax/lax_parallel.py

```diff
@@ -601,6 +604,10 @@ def all_gather(x, axis_name):
     x: array(s) with a mapped axis named ``axis_name``.
     axis_name: hashable Python object used to name a pmapped axis (see the
       :func:`jax.pmap` documentation for more details).
+    axis_index_groups: optional list of lists containing axis indices (e.g. for
+      an axis of size 4, [[0, 1], [2, 3]] would run all gather over the first
+      two and last two replicas). Groups must cover all axis indices exactly
+      once, and all groups must be the same size.
```

Comment:
While you are changing this, I think we should change below "with 2 XLA devices available" to "with 4 XLA devices available".

Reply:
Done.